packages/block-library/src/navigation/index.php

```diff
@@ -148,10 +162,39 @@ function render_block_core_navigation( $attributes, $content, $block ) {
 		)
 	);
 
+	$modal_unique_id = uniqid();
+
+	if ( ! isset( $attributes['isResponsive'] ) || false === $attributes['isResponsive'] ) {
+		return sprintf(
+			'<nav %1$s><ul class="wp-block-navigation__container">%2$s</ul></nav>',
+			$wrapper_attributes,
+			$inner_blocks_html
+		);
+	}
+
+	// Determine whether or not navigation elements should be wrapped in the markup required to make it responsive.
+	$responsive_container_markup = sprintf(
+		'<button aria-expanded="false" aria-haspopup="true" aria-label="%3$s" class="wp-block-navigation__responsive-container-open" data-micromodal-trigger="modal-%2$s"><svg width="24" height="24" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" role="img" aria-hidden="true" focusable="false"><rect x="4" y="7.5" width="16" height="1.5" /><rect x="4" y="15" width="16" height="1.5" /></svg></button>
+			<div class="wp-block-navigation__responsive-container" id="modal-%2$s" aria-hidden="true">
+				<div class="wp-block-navigation__responsive-close" tabindex="-1" data-micromodal-close>
+					<div class="wp-block-navigation__responsive-dialog" role="dialog" aria-modal="true" aria-labelledby="modal-%2$s-title" >
+							<button aria-label="%4$s" data-micromodal-close class="wp-block-navigation__responsive-container-close"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" width="24" height="24" role="img" aria-hidden="true" focusable="false"><path d="M13 11.8l6.1-6.3-1-1-6.1 6.2-6.1-6.2-1 1 6.1 6.3-6.5 6.7 1 1 6.5-6.6 6.5 6.6 1-1z"></path></svg></button>
+						<div class="wp-block-navigation__responsive-container-content" id="modal-%2$s-content">
+							<ul class="wp-block-navigation__container">%1$s</ul>
+						</div>
+					</div>
+				</div>
+			</div>',
+		$inner_blocks_html,
```

Comment:
Nitpick: I'd expect the order of arguments to be the same with the order of consumption for better following the code. It can be done in the follow ups though, not a blocker.